xarray/core/variable.py

```diff
@@ -199,6 +199,14 @@ def as_compatible_data(data, fastpath=False):
         # can't use fastpath (yet) for scalars
         return _maybe_wrap_data(data)
 
+    try:
+        from dask.dataframe.core import DataFrame as ddf
+
+        if isinstance(data, ddf):
+            return data.to_dask_array(lengths=True)
+    except ModuleNotFoundError:
+        print("Dask is not installed in your environment")
```

Comment:
could you move the try / except to `xarray.core.pycompat`? That way, `as_compatible_data` is a little bit more readable:
```suggestion
    if isinstance(data, dask_dataframe_type):
        return data.to_dask_array(lengths=True)
```